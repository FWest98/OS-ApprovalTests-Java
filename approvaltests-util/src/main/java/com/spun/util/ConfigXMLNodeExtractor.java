package com.spun.util;

import com.spun.util.io.XMLNodeExtractor;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import java.util.ArrayList;
import java.util.HashMap;

public class ConfigXMLNodeExtractor implements XMLNodeExtractor
{
  public void extractProperty(Node node, HashMap<String, Object> properties)
  {
    //String name = node.getNodeName();
    if (XMLNodeExtractor.Utils.extractSingleton(node, properties))
    {
      // do nothing
    }
    else if (node.getChildNodes().getLength() > 0)
    {
      NodeList childNodes = node.getChildNodes();
      ArrayList<String> v = new ArrayList<String>();
      for (int i = 0; i < childNodes.getLength(); i++)
      {
        if (childNodes.item(i).getNodeName().equals("VALUE"))
        {
          v.add(StringUtils.loadNullableString(childNodes.item(i).getChildNodes().item(0).getNodeValue()));
        }
      }
      properties.put(node.getNodeName(), StringUtils.toArray(v));
    }
  }
}
