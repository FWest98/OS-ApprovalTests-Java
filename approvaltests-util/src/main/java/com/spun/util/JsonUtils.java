package com.spun.util;

import com.google.gson.*;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import org.lambda.functions.Function1;

import java.io.IOException;
import java.time.Instant;

public class JsonUtils
{
  public static String prettyPrint(String json)
  {
    if (!ObjectUtils.isClassPresent("com.google.gson.Gson"))
    {
      throw new RuntimeException(
          "Missing Gson dependency\n  Pretty print uses Gson parser.\n  You can get this from the maven repo \n  or https://github.com/google/gson");
    }
    try
    {
      Gson gson = new GsonBuilder().setPrettyPrinting().create();
      JsonElement je = JsonParser.parseString(json);
      return gson.toJson(je);
    }
    catch (JsonSyntaxException e)
    {
      return String.format("Error:%s\nJson:\n%s", e.getMessage(), json);
    }
  }
  public static String asJson(Object o)
  {
    return asJson(o, (b) -> b);
  }
  public static <T> String asJsonWithBuilder(Object o, Function1<T, T> gsonBuilder, Class<T> gsonBuilderClass)
  {
    if (!ObjectUtils.isThisInstanceOfThat(gsonBuilderClass, com.google.gson.GsonBuilder.class))
    { throw new FormattedException("Class must be of type %s", GsonBuilder.class.getName()); }
    return asJson(o, (Function1<GsonBuilder, GsonBuilder>) gsonBuilder);
  }
  public static <T> String asJson(Object o, Function1<GsonBuilder, GsonBuilder> gsonBuilder)
  {
    GsonBuilder builder = new GsonBuilder().setPrettyPrinting();
    builder = addHandlingForDateObjects(builder);
    Gson gson = gsonBuilder.call(builder).create();
    return gson.toJson(o);
  }
  private static GsonBuilder addHandlingForDateObjects(GsonBuilder builder)
  {
    builder = builder.registerTypeAdapter(Instant.class, new InstantAdapter());
    return builder;
  }
  public static class InstantAdapter extends TypeAdapter<Instant>
  {
    @Override
    public void write(JsonWriter jsonWriter, Instant instant) throws IOException
    {
      jsonWriter.value(instant.toString());
    }
    @Override
    public Instant read(JsonReader jsonReader) throws IOException
    {
      throw new IOException("Never called");
    }
  }
}
