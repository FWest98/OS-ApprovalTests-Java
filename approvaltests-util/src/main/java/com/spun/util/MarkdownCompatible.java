package com.spun.util;

public interface MarkdownCompatible
{
  String toMarkdown();
}
