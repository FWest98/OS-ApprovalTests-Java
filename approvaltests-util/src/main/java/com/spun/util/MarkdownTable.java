package com.spun.util;

import org.lambda.functions.Function1;
import org.lambda.query.Queryable;

public class MarkdownTable implements MarkdownCompatible
{
  protected String markdown;
  public static <I, O> MarkdownTable create(I[] inputs, Function1<I, O> o, String column1, String column2)
  {
    MarkdownTable table = new MarkdownTable().withColumnHeaders(column1, column2);
    for (I input : inputs)
    {
      table.addRow(input, o.call(input));
    }
    return table;
  }
  public static MarkdownTable withHeaders(String... columnNames)
  {
    MarkdownTable table = new MarkdownTable();
    return table.withColumnHeaders(columnNames);
  }
  public <I> MarkdownTable addRowsForInputs(I[] inputs, Function1<I, Object>... transfers)
  {
    for (I input : inputs)
    {
      Queryable<Object> row = Queryable.as(transfers).select(f -> f.call(input));
      row.add(0, input);
      addRow(row.toArray());
    }
    return this;
  }
  public MarkdownTable addRow(Object... columns)
  {
    markdown += printRow(columns);
    return this;
  }
  public MarkdownTable withColumnHeaders(String... headers)
  {
    markdown = printColumnHeaders(headers);
    return this;
  }
  @Override
  public String toString()
  {
    return toMarkdown();
  }
  @Override
  public String toMarkdown()
  {
    return markdown;
  }
  public static String printColumnHeaders(String... headers)
  {
    return printRow(headers) + printRow(ArrayUtils.of("---", headers.length));
  }
  public static String printRow(Object... columns)
  {
    StringBuffer b = new StringBuffer();
    b.append("|");
    for (int x = 0; x < columns.length; ++x)
    {
      b.append(String.format(" %s |", columns[x]));
    }
    b.append("\n");
    return b.toString();
  }
}
