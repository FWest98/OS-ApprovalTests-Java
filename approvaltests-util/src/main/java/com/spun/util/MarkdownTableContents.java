package com.spun.util;

public class MarkdownTableContents implements MarkdownTableElement, Resizable
{
  private String contents;
  private int    padUntil;
  public MarkdownTableContents(String contents)
  {
    this.contents = contents;
  }
  @Override
  public String toString()
  {
    return " " + StringUtils.pad(contents, padUntil) + " ";
  }
  public int getLength()
  {
    return contents.length();
  }
  public void setPadding(int length)
  {
    padUntil = length;
  }
}
