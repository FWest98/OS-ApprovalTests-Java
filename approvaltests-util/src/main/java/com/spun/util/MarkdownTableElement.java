package com.spun.util;

public interface MarkdownTableElement {
    MarkdownTableElement DELIMITER = new MarkdownTableElement() {
        @Override
        public String toString() {
            return "|";
        }
    };
    MarkdownTableElement NEWLINE = new MarkdownTableElement() {
        @Override
        public String toString() {
            return "\n";
        }
    };
}
