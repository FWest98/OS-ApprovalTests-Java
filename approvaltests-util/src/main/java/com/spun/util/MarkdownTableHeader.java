package com.spun.util;

public class MarkdownTableHeader implements MarkdownTableElement, Resizable
{
  private int padUntil = 1;
  public MarkdownTableHeader()
  {
  }
  @Override
  public String toString()
  {
    int dashes = padUntil + 2; // Math.max(3, padUntil + 2);
    return "----------------------------------".substring(0, dashes);
  }
  public int getLength()
  {
    return 1;
  }
  public void setPadding(int length)
  {
    padUntil = length;
  }
}
