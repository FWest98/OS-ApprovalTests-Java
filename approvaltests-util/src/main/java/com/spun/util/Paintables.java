package com.spun.util;

import com.spun.swing.Paintable;

import java.awt.Graphics;
import javax.swing.JPanel;

public class Paintables
{
  public static JPanel asPanel(Paintable paintable)
  {
    return new JPanel()
    {
      {
        setPreferredSize(paintable.getSize());
      }
      @Override
      public void paint(Graphics g)
      {
        paintable.paint(g);
      }
    };
  }
}
