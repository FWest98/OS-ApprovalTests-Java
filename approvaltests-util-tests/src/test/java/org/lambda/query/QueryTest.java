package org.lambda.query;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class QueryTest
{
  @Test
  public void testQueryAll() throws Exception
  {
    List<Integer> numbers = Arrays.asList(3, 5, 7, 9);
    assertEquals(true, Query.all(numbers, n -> 3 <= n));
    assertEquals(false, Query.all(numbers, n -> 4 <= n));
  }
  @Test
  public void testQueryAllOfNothing() throws Exception
  {
    // None of the cases return false for empty data sets
    List<Integer> numbers = Arrays.asList();
    assertEquals(true, Query.all(numbers, n -> 3 <= n));
  }
}
