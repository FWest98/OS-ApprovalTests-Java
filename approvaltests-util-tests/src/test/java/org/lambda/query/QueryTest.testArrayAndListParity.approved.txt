Methods without a corresponding array or list


Query.all(Iterable,Function1)
Query.all(Object[],Function1)
Query.any(Iterable,Function1)
Query.any(Object[],Function1)
Query.average(Iterable,Function1)
Query.distinct(Iterable)
Query.first(Iterable,Function1)
Query.first(Object[],Function1)
Query.getTop(Iterable,Function1,int)
Query.getTop(Object[],Function1,int)
Query.groupBy(Iterable,Function1)
Query.groupBy(Iterable,Function1,Function1,Function1)
Query.lambda$groupBy$5(Queryable)
Query.last(Iterable)
Query.last(Object[])
Query.max(Iterable)
Query.max(Iterable,Function1)
Query.max(Number[])
Query.max(Object[],Function1)
Query.min(Iterable)
Query.min(Iterable,Function1)
Query.orderBy(List,Function1)
Query.orderBy(List,Order,Function1)
Query.orderBy(Object[],Function1)
Query.orderBy(Object[],Order,Function1)
Query.select(Iterable,Function1)
Query.select(Object[],Function1)
Query.selectMany(Iterable,Function1)
Query.selectManyArray(Iterable,Function1)
Query.skip(Iterable,int)
Query.skip(Object[],int)
Query.sum(Iterable)
Query.sum(Iterable,Function1)
Query.sum(Number[])
Query.sum(Object[],Function1)
Query.take(Iterable,int)
Query.take(Object[],int)
Query.where(Iterable,Function1)
Query.where(Object[],Function1)
